app/services/course/assessment/question/koditsu_question_service.rb

```diff
@@ -78,11 +78,70 @@ def build_question_object
         templates: [{
           path: koditsu_programming_language_map[@type][:filename],
           content: @metadata['submission'],
-          prefix: @metadata['prepend'],
-          suffix: koditsu_programming_language_map[@type][:language] == 'cpp' ? '' : @metadata['append']
+          prefix: truncate_google_test_framework_and_clean_comments(@metadata['prepend']),
+          suffix: truncate_google_test_framework_and_clean_comments(@metadata['append'])
         }],
         exprTestcases: @test_cases
       }]
     }
   end
+
+  def clean_from_comments(snippet)
+    no_single_line_comments_snippet = snippet.gsub(/\/\/.*$/, '')
+    no_multiple_line_comments_snippet = no_single_line_comments_snippet.gsub(/\/\*.*?\*\//m, '')
+
+    no_multiple_backspaces = no_multiple_line_comments_snippet.
+                             gsub(/\r/, '').
+                             gsub(/\n\s{4}/, "\n").
+                             gsub(/\n{2,}/, "\n")
```

Comment:
What's the point of this? And what are "backspaces"?  
Remove this set of gsub if not critical.

Reply:
this is just for aesthetic purpose i.e. if we don't have this, then the content will merely replace those comments with multiple newline char, making the function looks ugly

It will return something like this:
```
class GlobalEnv:






   return 'something'
```

Reply:
It's a matter of naming convention. If the function says it cleans comments, then it should _only_ clean comments.